OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoLogic.cs

```diff
@@ -38,13 +50,22 @@ public GameInfoLogic(Widget widget, World world, IngameInfoPanel activePanel)
 				objectivesTabButton.OnClick = () => activePanel = IngameInfoPanel.Objectives;
 				objectivesTabButton.IsHighlighted = () => activePanel == IngameInfoPanel.Objectives;
 
+				var panel = hasError ? "SCRIPT_ERROR_PANEL" : iop.PanelName;
 				var objectivesPanel = widget.Get<ContainerWidget>("OBJECTIVES_PANEL");
 				objectivesPanel.IsVisible = () => activePanel == IngameInfoPanel.Objectives;
 
-				Game.LoadWidget(world, iop.PanelName, objectivesPanel, new WidgetArgs());
+				Game.LoadWidget(world, panel, objectivesPanel, new WidgetArgs());
 
 				if (activePanel == IngameInfoPanel.AutoSelect)
 					activePanel = IngameInfoPanel.Objectives;
+
+				Game.BeforeGameStart += ResetVideoPlayed;
+				string video = null;
+				if (world.LocalPlayer.WinState != WinState.Undefined)
```

Comment:
This will play the video immediately when the player win/lose, breaking cooperative missions/multiplayer - teams can still win after an individual player has lost the game.